common/vrnetlab.py

```diff
@@ -296,31 +311,56 @@ def create_tc_tap_ifup(self):
     def gen_mgmt(self):
         """Generate qemu args for the mgmt interface(s)"""
         res = []
-        # mgmt interface is special - we use qemu user mode network
         res.append("-device")
         mac = (
             "c0:00:01:00:ca:fe"
             if getattr(self, "_static_mgmt_mac", False)
             else gen_mac(0)
         )
         res.append(self.nic_type + f",netdev=p00,mac={mac}")
-        res.append("-netdev")
-        res.append(
-            "user,id=p00,net=10.0.0.0/24,"
-            "tftp=/tftpboot,"
-            "hostfwd=tcp:0.0.0.0:22-10.0.0.15:22,"  # ssh
-            "hostfwd=udp:0.0.0.0:161-10.0.0.15:161,"  # snmp
-            "hostfwd=tcp:0.0.0.0:830-10.0.0.15:830,"  # netconf
-            "hostfwd=tcp:0.0.0.0:80-10.0.0.15:80,"  # http
-            "hostfwd=tcp:0.0.0.0:443-10.0.0.15:443,"  # https
-            "hostfwd=tcp:0.0.0.0:9339-10.0.0.15:9339,"  # iana gnmi/gnoi
-            "hostfwd=tcp:0.0.0.0:57400-10.0.0.15:57400,"  # nokia gnmi/gnoi
-            "hostfwd=tcp:0.0.0.0:6030-10.0.0.15:6030,"  # gnmi/gnoi arista
-            "hostfwd=tcp:0.0.0.0:32767-10.0.0.15:32767,"  # gnmi/gnoi juniper
-            "hostfwd=tcp:0.0.0.0:8080-10.0.0.15:8080"  # sonic gnmi/gnoi, other http apis
-        )
+
+        if self.mgmt_nic_passthrough:
+            # mgmt interface is passthrough - we just create a normal mirred tap interface
+            if self.conn_mode == "tc":
+                res.append("-netdev")
+                res.append("tap,id=p00,ifname=tap0,script=/etc/tc-tap-ifup,downscript=no")
```

Comment:
Is it not possible to rewrite the rules in tc-tap-ifup to be more specific, so that it becomes possible to forward some traffic to the container itself, e.g. to the serial console port in range 5000-5099 or so?

Reply:
We'll need a separate management tap script for this, but I think this is also doable by adding another tc filter rule for a pass traffic for a specific range of ports instead of performing the mirred action.

Reply:
Right yes, modifying or adding a new script was what I had in mind.

You listed things that this PR breaks. I think downloads etc from the container was never really part of the promise, like the container itself serves no other purpose than to run the VM, but the serial console port is IMHO a breaking change. If it can be fixed, that's at least one less breaking change.